services/rpcfilters/api.go

```diff
@@ -114,6 +117,51 @@ func (api *PublicAPI) NewPendingTransactionFilter() rpc.ID {
 
 }
 
+// NewFilter creates a new filter and returns the filter id. It can be
+// used to retrieve logs when the state changes. This method cannot be
+// used to fetch logs that are already stored in the state.
+//
+// Default criteria for the from and to block are "latest".
+// Using "latest" as block number will return logs for mined blocks.
+// Using "pending" as block number returns logs for not yet mined (pending) blocks.
+// In case logs are removed (chain reorg) previously returned logs are returned
+// again but with the removed property set to true.
```

Comment:
This comment comes from the implementation of logFilter in go-ethereum at `vendor/github.com/ethereum/go-ethereum/eth/filters/api.go`. Still a bit confused about the params in FilterCriteria:
 - When we use "latest", we should return logs for mined blocks. Does this mean that the filter will return the logs for all blocks mined while the filter is active? Or is this the meaning of "pending"?
 - If the range of `fromBlock - toBlock` is already mined, should we just add all of the logs for that range immediately and close the channel?
 - If the range is `fromBlock - pending/latest`, what should the behavior be?

Reply:
> When we use "latest", we should return logs for mined blocks. Does this mean that the filter will return the logs for all blocks mined while the filter is active? Or is this the meaning of "pending"?

Yes, but when user request filter's changes, filter cleans itself up, so it is not an ever-growing list. I'm not really sure how it should work with `pending` blocks, maybe you can try observing the behaviour in MetaMask or local go-ethereum node for that...

> If the range of fromBlock - toBlock is already mined, should we just add all of the logs for that range immediately and close the channel?

I'm not sure what the behaviour here should be, it's either what you've described or just empty filter. Again, I'll recommend you to launch a local go-ethereum node, connect it to a test network and observe the behaviour there. I don't have a better answer :(

> If the range is fromBlock - pending/latest, what should the behavior be?
 
It should probably only allow `latest` to `pending` or `latest` to `latest` or `pending` to `pending` ranges.
But to ensure the best compatibility we probably should do the same as Metamask does. A simple JS app using web3 library can answer all these questions :)